src/domain/service/search.ts

```diff
@@ -0,0 +1,276 @@
+      { $match: query },
+      {
+        $group: {
+          _id: '$state',
+          count: { $sum: 1 },
+        },
+      },
+    ]);
+    aggByStateResult.forEach((d) => {
+      countByState[d._id as State] = d.count;
+    });
+  }
+  const copy = apps.map(
+    (app: ApplicationDocument) =>
+      ({
+        appId: `${app.appId}`,
+        applicant: { info: app.sections.applicant.info, address: app.sections.applicant.address },
+        submitterId: app.submitterId,
+        approvedAtUtc: app.approvedAtUtc,
+        closedAtUtc: app.closedAtUtc,
+        closedBy: app.closedBy,
+        expiresAtUtc: app.expiresAtUtc,
+        state: app.state,
+        ethics: {
+          // tslint:disable-next-line:no-null-keyword
+          declaredAsRequired: app.sections.ethicsLetter.declaredAsRequired,
+        },
+        submittedAtUtc: app.submittedAtUtc,
+        lastUpdatedAtUtc: app.lastUpdatedAtUtc,
+        attestedAtUtc: app.attestedAtUtc,
+        isAttestable: isAttestable(app),
+        ...(params.includeCollaborators && {
+          collaborators: app.sections.collaborators.list.map((collab: Collaborator) => collab.info),
+        }),
+        ableToRenew: isRenewable(app),
+        revisionsRequested: wasInRevisionRequestState(app),
+        currentApprovedAppDoc: !!app.approvedAppDocs.find((doc) => doc.isCurrent),
+        ...(app.approvedAtUtc && {
+          attestationByUtc: getAttestationByDate(app.approvedAtUtc),
+        }),
+        lastPausedAtUtc: getLastPausedAtDate(app),
+        isRenewal: app.isRenewal,
+      } as ApplicationSummary),
+  );
+
+  return {
+    pagingInfo: {
+      totalCount: count,
+      pagesCount: Math.ceil((count * 1.0) / params.pageSize),
+      index: params.page,
+    },
+    items: copy,
+    stats: params.includeStats
+      ? {
+          countByState: countByState,
+        }
+      : undefined,
+  };
+}
+
+export const searchCollaboratorApplications = async (identity: Identity) => {
+  // find all applications on which the logged-in user has collaborator access
+  // using ego token email matched against collaborator googleEmail
+  const apps = await ApplicationModel.find({
+    state: 'APPROVED',
+    'sections.collaborators.list.info.googleEmail': identity.tokenInfo.context.user.email,
+  });
+
+  return apps.map(
+    (app: ApplicationDocument) =>
+      ({
+        appId: `${app.appId}`,
+        applicant: { info: app.sections.applicant.info },
+        expiresAtUtc: app.expiresAtUtc,
+      } as Partial<ApplicationSummary>),
+  );
+};
+
+export const getApplicationUpdates = async () => {
+  // do not return empty arrays. this is for apps existing before reset_updates_list migration
+  // this state should not be possible for applications created after this migration
+  const apps = await ApplicationModel.find(
+    { updates: { $ne: [] } },
+    { appId: 1, updates: 1 },
+  ).exec();
+
+  return apps;
+};
+
+export async function deleteApp(id: string, identity: Identity) {
```

Comment:
should this be in application service?

Reply:
yeah i wasn't sure where to put this one! prolly makes more sense to be there